boards/opentitan/src/tests/otbn.rs

```diff
@@ -136,29 +136,38 @@ fn otbn_run_rsa_binary() {
         let slice = unsafe { core::slice::from_raw_parts_mut(dmem_start as *mut u8, dmem_length) };
 
         CALLBACK.reset();
-        assert_eq!(otbn.load_data(0, slice), Ok(()));
+        assert_eq!(otbn.load_data(0, LeasableBuffer::new(slice)), Ok(()));
         run_kernel_op(1000);
 
         // Set the RSA mode to encryption
         // The address is the offset of `mode` in the RSA elf
         unsafe {
             TEMP[0] = 1;
-            assert_eq!(otbn.load_data(0, &mut TEMP[0..4]), Ok(()));
+            assert_eq!(
+                otbn.load_data(0, LeasableBuffer::new(&mut TEMP[0..4])),
+                Ok(())
+            );
         }
         run_kernel_op(1000);
 
         // Set the RSA length
         // The address is the offset of `n_limbs` in the RSA elf
         unsafe {
             TEMP[0] = (MODULUS.len() / 32) as u8;
-            assert_eq!(otbn.load_data(4, &mut TEMP[0..4]), Ok(()));
+            assert_eq!(
+                otbn.load_data(4, LeasableBuffer::new(&mut TEMP[0..4])),
+                Ok(())
+            );
```

Comment:
This isn't really a comment on this PR, since this code already did this before, but this is probably UB -- you're taking multiple  `&'static mut` references to the TEMP buffer and passing them to callees. Given what the underlying code does and that this is single threaded it probably will do the correct thing in practice, and this is just test code. Still, having multiple static buffers (one for each use), and initializing them using static_init!() would probably be safer.

Reply:
That's probably a good idea. I'll work on this